site/src/components/introduction/building-a-chart.md

```diff
@@ -0,0 +1,188 @@
+{{{ dynamic-include 'javascript' js="creating-some-data-js" }}}
+
+## Rendering the data
+
+The sales data is rendered as a bar chart. With D3 this typically involves rendering paths or rectangles and doing a bit of maths to locate each bar correctly, as illustrated in [Mike Bostock's excellent tutorial](http://bost.ocks.org/mike/bar/).
+
+d3fc has a [bar series component](/components/series/bar.html) (among others) that makes it much easier to render your data. To render the data, we create a bar series component and supply it with suitable D3 scales, configure the `xValue` and `yValue` accessors (which the component uses to extract the x and y values from the datapoints).
+
+The bar series is a regular D3 component, and is rendered via the `call` method on a D3 selection:
+
+```js
+{{{ codeblock rendering-a-series-js }}}
+```
+
+This renders as follows:
+
+{{{ dynamic-include 'codepen' html="rendering-a-series-html" js="rendering-a-series-js"}}}
+{{{ rendering-a-series-html }}}
+{{{ dynamic-include 'javascript' js="rendering-a-series-js" }}}
+
+At this point you might like to read more about the [D3 component pattern](http://bost.ocks.org/mike/chart/) that we follow with d3fc.
+
+I'll forgive you if you are not too impressed with the above example. So far you've got a bar series, but no axes or labels, yet the example involves quite a bit of code.
+
+The reason for this is that the bar series component is relatively low-level component, it renders a series and nothing more. (It's actually not the lowest level of component, the bar series uses the d3fc bar SVG component). In order to quickly assemble a chart you need to make use of a more high-level component ... a chart.
+
+## Cartesian chart
+
+The cartesian chart component does a number of things, it measures its container in order to fill it, setting a suitable range on the scales that you supply (it's still your job to set the domain on each). The chart renders a pair of axes and, optionally, other cosmetic features such as chart and axis labels.
+
+You can render a series within the chart by setting it as the `plotArea`. The chart sets the x and y scales on the component you supply.
+
+Here's the complete example:
+
+```js
+{{{ codeblock cartesian-chart-js }}}
+```
+
+This renders a more complete chart:
+
+{{{ dynamic-include 'codepen' html="cartesian-chart-html" js="cartesian-chart-js"}}}
+{{{ cartesian-chart-html }}}
+
+{{{ dynamic-include 'javascript' js="cartesian-chart-js" }}}
+
+## Setting the axis domain
+
+The current y-domain is a bit of a guess, being hard-coded to a range `[0, 20]`. Ideally the domain should be computed based on the data. In this case the domain should accommodate the greatest sales value, plus some padding, and also extent to zero.
+
+There's another d3fc component that can help here, the [extent](/components/util/extent.html) utility. The following code ensures that zero is included, and adds a small amount of padding to the 'top' end of the range:
```

Comment:
I wasn't sure what type of padding was added (percentage / pixel) until I looked at the source.